aws_lambda_powertools/utilities/idempotency/base.py

```diff
@@ -227,6 +230,11 @@ def _handle_for_status(self, data_record: DataRecord) -> Optional[Any]:
             )
         response_dict: Optional[dict] = data_record.response_json_as_dict()
         if response_dict is not None:
+            if self.config.response_hook is not None:
+                return self.config.response_hook(
+                    self.output_serializer.from_dict(response_dict),
+                    IdempotentHookData(data_record),
+                )
             return self.output_serializer.from_dict(response_dict)
```

Comment:
Suggestion just to keep the code cleaner

```suggestion
            serialized_output = self.output_serializer.from_dict(response_dict)
            if self.config.response_hook is not None:
                return self.config.response_hook(
                    serialized_output
                    IdempotentHookData(data_record),
                )
            return serialized_output
```